src/GeositeFramework/js/Plugin.js

```diff
@@ -496,58 +506,59 @@ require(['use!Geosite',
-                    _.delay(parseDeferred.resolve, 200);
-
-                    var mapReadyDeferred = setupPrintableMap(view.model.get('pluginObject').map, pluginObject, $printSandbox, previewDeferred);
-
-                    mapReadyDeferred.then(function(previewMap) {
-                        // The plugin is given a deferred object to resolve when the page is ready
-                        // to be printed, a reference to an element where it can place printable
-                        // elements outside of its container and a reference to an esriMap which
-                        // is used as a print preview box.
-                        pluginObject.beforePrint(pluginDeferred, $printSandbox, previewMap);
-
-                        // Exectue the browser print when the plugin and print preview (if used)
-                        // have responded, as well as a slight delay for css parsing.
-                        $.when(pluginDeferred, parseDeferred, previewDeferred).then(function() {
-                            window.print();
-                        });
-                    });
-                }).end()
-                .hide();
+                    initPrint(model, paneNumber);
+                }).end();
+        }
 
-            // Attach to top pane element
-            view.$el.parents().find('.content .nav-apps').after($uiContainer);
+        function initPrint(model, paneNumber) {
+            var pluginDeferred = $.Deferred(),
+                parseDeferred = $.Deferred(),
+                previewDeferred = $.Deferred(),
+                pluginCssPath = model.get('pluginSrcFolder') + '/print.css',
+                printCssClass = 'plugin-print-css',
+                oppositePaneHideCssPath = 'css/print-hide-map' +
+                    (paneNumber === 0 ? 1 : 0) + '.css',
+                $printSandbox = $('#plugin-print-sandbox'),
+                pluginObject = model.get('pluginObject');
 
-            // Tell the model about $uiContainer so it can pass it to the plugin object
-            model.set('$uiContainer', $uiContainer);
+            // Any previous plugin-prints may have left specific print css
+            // or sandbox elements.  Clear all so that this new print routine
+            // has no conflicts with other plugins.
+            $('.' + printCssClass).remove();
+            $printSandbox.empty();
+
+            // add base plugin css
+            addCss('css/print.css', 'base-plugin-print-css');
+
+            // Add the plugin css
+            addCss(pluginCssPath, printCssClass);
+
+            // Hide the possible same plugin from the opposite map pane
+            addCss(oppositePaneHideCssPath, printCssClass);
+
+            // Give some time for the browser to parse the new CSS,
+            // if this wasn't slightly delayed, occasionally the override would
+            // not be present and print features wouldn't show up.
+            _.delay(parseDeferred.resolve, 200);
```

Comment:
Would this work with a delay value of `0`? Wondering if the `200` value is significant here or if it's just that this needs to be pushed to the back of the event loop queue.

Reply:
This line is showing up new because of how things were moved around, but I don't know for sure. I imagine that using `0` didn't work in every case.